src/btree/row_srch.c

```diff
@@ -132,6 +132,67 @@ __wt_search_insert(
 }
 
 /*
+ * __check_leaf_key_range --
+ *	Check the search key is in the leaf page's key range.
+ */
+static inline int
+__check_leaf_key_range(WT_SESSION_IMPL *session,
+    WT_ITEM *srch_key, WT_REF *leaf, WT_CURSOR_BTREE *cbt)
+{
+	WT_BTREE *btree;
+	WT_COLLATOR *collator;
+	WT_ITEM *item;
+	WT_PAGE_INDEX *pindex;
+	uint32_t indx;
+	int cmp;
+
+	btree = S2BT(session);
+	collator = btree->collator;
+	item = cbt->tmp;
+
+	/*
+	 * Check if the search key is less than the parent's starting key for
+	 * this page.
+	 */
+	__wt_ref_key(leaf->home, leaf, &item->data, &item->size);
+	WT_RET(__wt_compare(session, collator, srch_key, item, &cmp));
+	if (cmp < 0) {
+		cbt->compare = 1;		/* page keys > search key */
+		return (0);
+	}
```

Comment:
I agree -- for some reason I've been thinking a fine-grained understanding of why we failed might be useful in the future, but I doubt that's really true. Might as well just return a boolean (since the column-store version of this function can't fail, so it simplifies it a lot).